src/main/java/org/opensearch/knn/quantization/factory/QuantizerRegistry.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.factory;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.opensearch.knn.quantization.models.quantizationParams.QuantizationParams;
+import org.opensearch.knn.quantization.quantizer.Quantizer;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+/**
+ * The QuantizerRegistry class is responsible for managing the registration and retrieval
+ * of quantizer instances. Quantizers are registered with specific quantization parameters
+ * and type identifiers, allowing for efficient lookup and instantiation.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+final class QuantizerRegistry {
+    // ConcurrentHashMap for thread-safe access
+    private static final Map<String, Supplier<? extends Quantizer<?, ?>>> registry = new ConcurrentHashMap<>();
+
+    /**
+     * Registers a quantizer with the registry.
+     *
+     * @param paramIdentifier   the unique identifier for the quantization parameters
+     * @param quantizerSupplier a supplier that provides instances of the quantizer
```

Comment:
Do we need to create a new quantizer for each request or can we reuse?

Reply:
Good Catch ....We can re-use, since Quantizer is stateless we can definitely do.